java/custom/com/nokia/mid/ui/gestures/GestureRegistrationManager.java

```diff
@@ -13,41 +15,77 @@ public ZoneRegistration(Object container, GestureInteractiveZone zone) {
+    }
+
+    private static GestureRegistrationManager gestureRegistrationManager = null;
+    private static GestureRegistrationManager getRegistrationManagerInstance() {
+        if (gestureRegistrationManager == null) {
+            gestureRegistrationManager = new GestureRegistrationManager();
         }
+
+        return gestureRegistrationManager;
     }
 
-    public static void callListener(GestureEvent event) {
+    public boolean preprocess(Event event, Event waitingEvent) {
+        return true;
+    }
+
+    public void process(Event event) {
+        NativeEvent nativeEvent = (NativeEvent)event;
+        GestureEvent gestureEvent = new GestureEventImpl(nativeEvent.intParam1,
+                                                         nativeEvent.intParam2,
+                                                         nativeEvent.intParam3,
+                                                         nativeEvent.intParam5,
+                                                         nativeEvent.intParam6,
+                                                         nativeEvent.floatParam1,
+                                                         nativeEvent.intParam7,
+                                                         nativeEvent.intParam8,
+                                                         nativeEvent.intParam9,
+                                                         nativeEvent.intParam10,
+                                                         nativeEvent.intParam11,
+                                                         nativeEvent.intParam12,
+                                                         nativeEvent.intParam13,
+                                                         nativeEvent.intParam14,
+                                                         nativeEvent.intParam15,
+                                                         nativeEvent.intParam16);
+
         for (int i = 0; i < zoneRegistrations.size(); i++) {
             ZoneRegistration zoneReg = (ZoneRegistration)zoneRegistrations.elementAt(i);
 
             GestureInteractiveZone zone = zoneReg.zone;
 
-            if (!zone.contains(event.getStartX(), event.getStartY())) {
+            if (!zone.contains(gestureEvent.getStartX(), gestureEvent.getStartY())) {
                 continue;
             }
 
-            if (!zone.supports(event.getType())) {
+            if (!zone.supports(gestureEvent.getType())) {
                 continue;
             }
 
             GestureListener listener = (GestureListener)listenerRegistrations.get(zoneReg.container);
             if (listener != null) {
-                listener.gestureAction(zoneReg.container, zone, event);
+                listener.gestureAction(zoneReg.container, zone, gestureEvent);
             }
         }
     }
 
+    public static void setListener(Object container, GestureListener listener) throws IllegalArgumentException {
```

Comment:
Why did this move? Can we move it back so it doesn't show up in the diff? Feel free to move it again in a follow-up patch